src/BenchmarkDotNet/Engines/Engine.cs

```diff
@@ -183,6 +183,13 @@ public Measurement RunIteration(IterationData data)
             // it does not matter, because we have already obtained the results!
             EnableMonitoring();
 
+            if (RuntimeInformation.IsNetCore)
```

Comment:
Since this problem is relevant only for .NET Core 3.x, could we add an additional check?